src/controllers/edgeconnect/deployment/deployment.go

```diff
@@ -57,7 +60,13 @@ func New(instance *edgeconnectv1alpha1.EdgeConnect) *appsv1.Deployment {
 }
 
 func prepareContainerEnvVars(instance *edgeconnectv1alpha1.EdgeConnect) []corev1.EnvVar {
-	envVars := []corev1.EnvVar{
+	var envVars []corev1.EnvVar
+
+	if len(instance.Spec.Env) > 0 {
+		envVars = instance.Spec.Env
+	}
+
+	mandatoryEnvVars := []corev1.EnvVar{
```

Comment:
the current way we handle user set envs is that they should take precedent, no matter what

we are going to tackle this topic later codebase wide(AND document the decision), if we want to be smarter about or not, but right now we should let the user envs take precedent